akka-http-core/src/main/scala/akka/http/scaladsl/settings/RoutingSettings.scala

```diff
@@ -36,7 +39,9 @@ abstract class RoutingSettings private[akka] () extends akka.http.javadsl.settin
   override def withRangeCountLimit(rangeCountLimit: Int): RoutingSettings = self.copy(rangeCountLimit = rangeCountLimit)
   override def withRangeCoalescingThreshold(rangeCoalescingThreshold: Long): RoutingSettings = self.copy(rangeCoalescingThreshold = rangeCoalescingThreshold)
   override def withDecodeMaxBytesPerChunk(decodeMaxBytesPerChunk: Int): RoutingSettings = self.copy(decodeMaxBytesPerChunk = decodeMaxBytesPerChunk)
-  override def withFileIODispatcher(fileIODispatcher: String): RoutingSettings = self.copy(fileIODispatcher = fileIODispatcher)
+  @deprecated("binary compatibility method. Use `akka.stream.materializer.blocking-io-dispatcher` to configure the dispatcher", since = "10.1.1")
+  @Deprecated
+  override def withFileIODispatcher(fileIODispatcher: String): RoutingSettings = self
```

Comment:
removal of the `file-io-dispatcher` config makes the whole `*FileIODispatcher` family of methods unnecessary, but they need to stay because of the binary compatibility. The implementation changes to no op.